src/App.tsx

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+enum FilterStatus {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<string>(FilterStatus.ALL);
+  const [query, setQuery] = useState<string>('');
+  const [isDataLoaded, setIsDataLoaded] = useState<boolean>(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodosFromServer(data);
+      })
+      .finally(() => setIsDataLoaded(true));
+  }, []);
```

Comment:
The `useEffect` dependency array is currently empty. To ensure the effect runs correctly when relevant values change, consider including dependencies such as `getTodos` if it's not a stable function reference.